main.js

```diff
@@ -1,6 +1,79 @@
-import './style.css'
+import axios from "axios";
+const key = 'f9d119ddc9f2ae7c8b2c34f1d5910a87';
+const api =`https://api.themoviedb.org/3/discover/movie?sort_by=popularity.desc&api_key=${key}`;
+const imgUrl ="https://image.tmdb.org/t/p/w500" ;
+const searchUrl= `https://api.themoviedb.org/3/search/movie?api_key=${key}`;
 
-document.querySelector('#app').innerHTML = `
-  <h1>Hello Vite!</h1>
-  <a href="https://vitejs.dev/guide/features.html" target="_blank">Documentation</a>
-`
+const app=document.querySelector("#app");
+const movieData=document.querySelector(".movie-card");
+const movieSearch=document.querySelector(".search-bar")
+const errorDiv=document.querySelector(".no-movie-found");
+
+
+async function getMovieFromApi(api) {
+	const response= await axios.get(api);
+	const movieData=response.data.results.slice(0,15);
+	if(movieData.length){
+		app.style.display="block";
+		showMovieData(movieData)
+		errorDiv.style.display="none";
+	}
+	else{
+		app.style.display="none";
+		errorDiv.style.display="block";
+
+	}
+}
+
+
+getMovieFromApi(api);
+
+movieSearch.addEventListener("keyup", (event)=>{
+	event.preventDefault();
+	const query=event.target.value;
+
+	if(query.length>0){
+		getMovieFromApi(searchUrl+"&query="+query);
```

Comment:
You can just pass query in this function since variables like api and searchUrl are globally scoped.
The getMovieFromApi function will be something like:
```
async function getMovieFromApi(query='') {
       const url = query==='' ? api : searchUrl+"&query="+query;
       const response = await axios.get(url);
       // rest of the code
}
```